CONTRIBUTING.md

```diff
@@ -0,0 +1,47 @@
+
+If you get stuck, reach out to the friendly folks in the `#pl-dev` channel on the [PrairieLearn Slack](https://go.illinois.edu/joinplslack)!
+
+## Opening a pull request
+
+We follow the [GitHub flow](https://docs.github.com/en/get-started/quickstart/github-flow) for all changes:
+
+- You should work on a distinct branch, not `master`. While this isn't strictly necessary for forks, it's helpful if you want to be working on multiple independent changes at the same time.
+- When committing your changes, use a short but meaningful commit message, e.g. `fix rate limiting` instead of `fix`.
+- Once you're happy with your changes, [open a pull request (PR)](https://docs.github.com/en/articles/creating-a-pull-request).
+  - You should include a reasonable amount of information with your pull request, such as a summary of what changes you made and why they were made.
+  - For changes that impact UI, it can be helpful to include screenshots and screen recordings.
+  - If applicable, include instructions on how to manually test or verify your change.
+  - If the change you're making will resolve an existing issue, you should [link the issue to the pull request](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) so that the issue is closed automatically once the pull request is merged.
+- Monitor the [GitHub status checks](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/collaborating-on-repositories-with-code-quality-features/about-status-checks). If they fail, you should check the output to see why. You may need to fix a broken test, adjust the formatting of your code, or fix a lint error.
+- Like many open-source projects, we require you to "sign" a contributor license agreement (CLA) prior to accepting any contributions. When you open your first pull request, a bot will prompt you to leave a comment stating that you accept the terms of [our CLA](https://github.com/PrairieLearn/cla).
+- One or more PrairieLearn maintainers will review your PR. You should be prepared to engage with the maintainers to answer questions, update code, etc.
+- Once the PR is in a satisfactory state, a maintainer will approve and merge your change! :tada:
+
+## Using your changes in production
+
+After your changes have been merged to `master`, a new [`prairielearn/prairielearn` Docker image](https://hub.docker.com/r/prairielearn/prairielearn) will be built and published to Docker Hub. If you're developing course content locally, you can then run `docker pull prairielearn/prairielearn` to access the new image.
+
+Usually, your changes will be deployed to all production instances of PrairieLearn within a week. Watch the `#announce` Slack channel to see when your change has been deployed. If it's been several weeks since your PR has been merged and your changes still haven't been deployed, feel free to reach out in the `#pl-dev` Slack channel.
```

Comment:
It may be useful to mention that the rate at which deployment is done may be impacted by typical academic calendars, e.g., there may be less deployments during exam seasons in December.

Reply:
Good idea, done in 579f036.